### PR TITLE
Add synchronous replication docs

### PR DESCRIPTION
**Pull request by CharLotteiu**
### What is changed, added or deleted? (Required)

Add a document for the synchronous replication. Content migrated from tikv/tikv#7740 and tikv/website#211.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/4207
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

File: synchronous-replication.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  }
+}
+```
+
+> **注意：**
+>
+> 集群的复制状态说明了 Region 被复制的方式，有三个选项：`async`， `sync-recover` 和 `sync`。
+
+集群的复制状态变为 `sync` 后，如果掉线的实例数大于任一数据中心应该存储的副本数，状态会自动变为 `async`。一旦集群的复制状态变为 `async`，PD 会要求 TiKV 将复制模式修改为 `asynchronous` 并不时检查 TiKV 实例是否已被恢复。而当掉线的实例数恢复到小于
+任一数据中心应该存储的副本数时，集群会进入 `sync-recover` 复制状态，并要求 TiKV 将复制模式修改为 `synchronous`。直到所有的 Region 都同步完毕后，集群会再次进入 `sync` 复制状态。
```
Comment (TomShawn) [inline synchronous-replication.md:67]:
```suggestion
集群的复制状态变为 `sync` 后，如果掉线的实例数大于任一数据中心应该存储的副本数，状态会自动变为 `async`。一旦集群的复制状态变为 `async`，PD 会要求 TiKV 将复制模式更改为 `asynchronous` 并时不时检查 TiKV 实例是否已被恢复。而当掉线的实例数恢复到小于任一数据中心应该存储的副本数时，集群会进入 `sync-recover` 复制状态，并要求 TiKV 将复制模式更改为 `synchronous`。直到所有的 Region 都同步完毕后，集群会再次进入 `sync` 复制状态。
```

**Review by TomShawn — COMMENTED** [review]
Minor comments on wording
@BusyJay PTAL

File: pd-control.md [hunk 2]
```diff
@@ -418,6 +418,8 @@ Leader 调度的开销较小，需要的时候可以适当调大。
     config set cluster-version 1.0.8
     ```
 
+- `replication-mode` 用于控制双数据中心场景下 Region 的复制模式，详见[手动更改复制模式](/synchronous-replication.md#手动更改复制模式)
```
Comment (CharLotteiu) [inline pd-control.md:421]:
```suggestion
- `replication-mode` 用于控制双数据中心场景下 Region 的复制模式，详见[手动更改复制模式](/synchronous-replication.md#手动更改复制模式)。
```

**Comment by TomShawn** [comment]
LGTM

**Comment by TomShawn** [comment]
After discussion with @BusyJay, we find that the feature name `synchronous replication` does not accurately describe this feature. @BusyJay is thinking of a new name and will update the English version first. So this PR is pending until the feature name is determined.

**Comment by TomShawn** [comment]
> After discussion with @BusyJay, we find that the feature name `synchronous replication` does not accurately describe this feature. @BusyJay is thinking of a new name and will update the English version first. So this PR is pending until the feature name is determined.

@BusyJay Any update?

**Comment by BusyJay** [comment]
The feature will be release in following sprints, when the docs can be updated.

**Comment by qiancai** [comment]
Hi @BusyJay, we are preparing the user docs for TiDB v5.1 and would like to know whether the doc changes of this PR apply to TiDB v5.1 or not. Would you please check and reply? Thanks!

**Comment by TomShawn** [comment]
@BusyJay Any update?

**Comment by TomShawn** [comment]
This PR is superseded by https://github.com/pingcap/docs-cn/pull/6843.

**Pull request closed without merging**